internal/stateindex/encoding.go

```diff
@@ -81,3 +98,35 @@ func decodeReverseOrderVarUint64(b []byte) uint64 {
 
 	return math.MaxUint64 - v
 }
+
+const hextable = "0123456789abcdef"
+
+// EncodedLen returns the length of an encoding of n source bytes.
+// Specifically, it returns n * 2.
+func encodedLen(n int) int { return n * 2 }
+
+// Encode encodes src into EncodedLen(len(src))
+// bytes of dst. As a convenience, it returns the number
+// of bytes written to dst, but this value is always EncodedLen(len(src)).
+// Encode implements hexadecimal encoding.
+func encode(dst, src []byte) int {
+	j := 0
+	for _, v := range src {
+		dst[j] = hextable[v>>4]
+		dst[j+1] = hextable[v&0x0f]
+		j += 2
+	}
+	return len(src) * 2
+}
+
+// fromHexChar converts a hex character into its value
+func fromHexChar(c byte) byte {
```

Comment:
as commented above, this may result in an error when the char is not `0-9` `a-f` `A-F`

Reply:
Yes, it was an intentional but changed as per your suggestion.